src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -115,34 +141,47 @@ public override bool Execute ()
 			if (CreatePackagePerAbi)
 				Log.LogDebugMessage ("  SupportedAbis: {0}", SupportedAbis);
 
-			bool upToDate = ManifestIsUpToDate (ManifestFile);
-
 			if (ResourceNameCaseMap != null)
 				foreach (var arr in ResourceNameCaseMap.Split (';').Select (l => l.Split ('|')).Where (a => a.Length == 2))
 					resource_name_case_map [arr [1]] = arr [0]; // lowercase -> original
 
-			if (AdditionalAndroidResourcePaths != null)
-				foreach (var dir in AdditionalAndroidResourcePaths)
-					if (!string.IsNullOrEmpty (dir.ItemSpec))
-						upToDate = upToDate && ManifestIsUpToDate (string.Format ("{0}{1}{2}", dir, Path.DirectorySeparatorChar, "manifest", Path.DirectorySeparatorChar, "AndroidManifest.xml"));
+			ThreadingTasks.Parallel.ForEach (ManifestFiles, () => 0, (manifestFile, loopState, localSum) => {
```

Comment:
Given that so much is being re-indented and moved around *anyway*, can we move this `(manifestFile, loopState, localSum)` lambda into a separate and distinct *method*? That would reduce the complexity of `Execute()`.